src/lambda_function.py

```diff
@@ -1,61 +1,82 @@
-import os
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this
+# software and associated documentation files (the "Software"), to deal in the Software
+# without restriction, including without limitation the rights to use, copy, modify,
+# merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+# INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+# PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
 import json
 import logging
+import os
 
 
 # Set LogLevel using environment variable, fallback to INFO if not present
 logger = logging.getLogger()
-try:
-    log_level = os.environ['LogLevel']
-    if log_level not in ['INFO', 'DEBUG']:
-        log_level = 'INFO'
-except:
+log_level = os.getenv('LogLevel', 'INFO')
+if log_level not in ['INFO', 'DEBUG']:
     log_level = 'INFO'
 logger.setLevel(log_level)
+global log_prefix 
+log_prefix = " - "
 
 # Load environment variables
-wav_bucket = os.environ['WAVFILE_BUCKET']
+wav_bucket = os.getenv('WAVFILE_BUCKET', None)
+
+
+# handler mapping table -- calls the method 'value' for the event 'key'
+handlers = {
+    'NEW_INBOUND_CALL': lambda call_id, event, participants: new_call_handler(call_id),
+    'RINGING': lambda call_id, event, participants: response(),
+    'ACTION_SUCCESSFUL': lambda call_id, event, participants: action_success_handler(call_id, event),
+    'DIGITS_RECEIVED': lambda call_id, event, participants: control_voicefocus(call_id, event),
+    'HANGUP': lambda call_id, event, participants: hangup(participants),
+    # 'ACTION_FAILED': lambda call_id, event, participants: (
```

Comment:
Testing Comments should be removed.

Reply:
agreed.  this PR should NOT really be merged... i wanted to stimulate discussion on the approach before completing the refactor.